benchmarks/src/main/java/org/elasticsearch/benchmark/xcontent/FilterContentBenchmark.java

```diff
@@ -0,0 +1,182 @@
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+@Fork(1)
+@Warmup(iterations = 2)
+@Measurement(iterations = 3)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+public class FilterContentBenchmark {
+
+    @Param({ "cluster_stats", "index_stats", "node_stats" })
+    private String type;
+
+    @Param({ "10_field", "half_field", "all_field", "wildcard_field", "10_wildcard_field" })
+    private String fieldCount;
+
+    @Param({ "true", "false" })
+    private boolean inclusive;
+
+    private BytesReference source;
+    private XContentParserConfiguration parserConfig;
+    private Set<String> filters;
+
+    @Setup
+    public void setup() throws IOException {
+        String sourceFile;
+        switch (type) {
+            case "cluster_stats":
+                sourceFile = "monitor_cluster_stats.json";
+                break;
+            case "index_stats":
+                sourceFile = "monitor_index_stats.json";
+                break;
+            case "node_stats":
+                sourceFile = "monitor_node_stats.json";
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown type [" + type + "]");
+        }
+        source = readSource(sourceFile);
+
+        Map<String, Object> flattenMap = Maps.flatten(XContentHelper.convertToMap(source, true, XContentType.JSON).v2(), false, true);
+        Set<String> keys = flattenMap.keySet();
+        switch (fieldCount) {
+            case "10_field": {
+                AtomicInteger count = new AtomicInteger();
+                Set<String> filterKeys = keys.stream()
+                    .filter(key -> count.getAndIncrement() % 5 == 0)
+                    .limit(10)
+                    .collect(Collectors.toSet());
+                filters = filterKeys;
+            }
```

Comment:
Could you drop the `{` and `}` here? The formatter spat out some pretty confusing indentation.

Reply:
It might be more clear to make it a method call that `return`s the filters instead of keep it all in the `setup` method.

Reply:
done